pkg/render/kubecontrollers/kube-controllers.go

```diff
@@ -106,29 +144,33 @@ func (c *kubeControllersComponent) Objects() ([]client.Object, []client.Object)
 		c.controllersDeployment(),
 	}
 	objectsToDelete := []client.Object{}
-	if c.cfg.ManagerInternalSecret != nil {
+	if c.kubeControllerName == KubeController && c.cfg.ManagerInternalSecret != nil {
 		objectsToCreate = append(objectsToCreate, secret.ToRuntimeObjects(
 			secret.CopyToNamespace(common.CalicoNamespace, c.cfg.ManagerInternalSecret)...)...)
 	}
 
-	if c.cfg.ElasticsearchSecret != nil {
-		objectsToCreate = append(objectsToCreate, secret.ToRuntimeObjects(
-			secret.CopyToNamespace(common.CalicoNamespace, c.cfg.ElasticsearchSecret)...)...)
-	}
+	if c.kubeControllerName == EsKubeController {
+		if c.cfg.ElasticsearchSecret != nil {
```

Comment:
I wonder if for these situations we can still make the decision of whether or not the es secret is rendered in the constructor. You could add a boolean field `renderElasticsearchSecret`, and set that to true in the `NewElasticsearchKubeControllers` if `ElasticsearchSecret` is not nil. Then this becomes:

```
if c.renderElasticsearchSecret {
    objectsToCreate = append(objectsToCreate, secret.ToRuntimeObjects(
				secret.CopyToNamespace(common.CalicoNamespace, c.cfg.ElasticsearchSecret)...)...)
}
```

My preference would be to follow this pattern and remove the `kubeControllerConfigName` variable.